src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,82 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const [user, setUser] = useState<User | null>();
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    setError('');
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(data => setUser(data))
+        .catch(() => setError('User not found'))
+        .finally(() => setIsLoading(false));
+    }
+  }, [currentTodo]);
+
+  if (!currentTodo) {
+    return;
```

Comment:
The `return` statement should return `null` instead of nothing to prevent rendering issues: `return null;`